packages/account/src/providers/provider.ts

```diff
@@ -164,6 +164,7 @@ export type TransactionCost = {
   requiredQuantities: CoinQuantity[];
   addedSignatures: number;
   dryRunStatus?: DryRunStatus;
+  setMaxFee?: boolean;
```

Comment:
I'd rename this to `updateMaxFee` because `setMaxFee` is already used in the context of a `maxFee` value that was _set_ e.g. [here](https://github.com/FuelLabs/fuels-ts/blob/st/fix/avoid-overriding-user-gaslimit-maxfee/packages/program/src/functions/base-invocation-scope.ts#L465) and in some other places.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2262/commits/0452091dd23ff2608de1954f787f8734565b4d65

Reply:
Sure, `isMaxFeeSet` sounds good as well.